### PR TITLE
fix: always allow metadata queries

### PR DESCRIPTION
**Pull request by olavloite**
Internal metadata queries should always be allowed, regardless of the type of transaction that is currently running or any special mode that has been set on the connection. Internal metadata queries are system queries that are generated by a connection to return metadata to an application, for example methods like getTables() in JDBC.

### CODE REVIEW

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java [hunk 1]
```diff
@@ -133,17 +133,6 @@ enum BatchMode {
     DML
   }
 
-  /**
```
Comment (olavloite) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java:136]: This was dead code. Only the class in the public `Connection` interface was used.

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/DdlBatch.java [hunk 2]
```diff
@@ -121,29 +118,6 @@ public ApiFuture<ResultSet> executeQueryAsync(
       final ParsedStatement statement,
       AnalyzeMode analyzeMode,
       QueryOption... options) {
-    if (options != null) {
```
Comment (olavloite) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/DdlBatch.java:124]: This special handling is no longer needed, as it is generically handled in the connection.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/DdlBatchTest.java [hunk 3]
```diff
@@ -157,25 +157,6 @@ public void testExecuteCreateDatabase() {
                     .parse(Statement.of("CREATE DATABASE foo"))));
   }
 
-  @Test
-  public void testExecuteMetadataQuery() {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/DdlBatchTest.java:161]: This test specifically checked whether a DDL batch could execute an internal metadata query. That is no longer needed, as a DDL batch no longer handles that special case itself. Instead, this is handled directly in the connection for all types of transactions.

Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/DdlBatchTest.java:161]: This test specifically checked whether a DDL batch could execute an internal metadata query. That is no longer needed, as a DDL batch no longer handles that special case itself. Instead, this is handled directly in the connection for all types of transactions.

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java [hunk 4]
```diff
@@ -1514,22 +1519,31 @@ private ApiFuture<long[]> internalExecuteBatchUpdateAsync(
...
+      this.currentUnitOfWork = createNewUnitOfWork(false);
     }
     return this.currentUnitOfWork;
   }
 
   @VisibleForTesting
-  UnitOfWork createNewUnitOfWork() {
-    if (isAutocommit() && !isInTransaction() && !isInBatch()) {
+  UnitOfWork createNewUnitOfWork(boolean isInternalMetadataQuery) {
+    if (isInternalMetadataQuery || isAutocommit() && !isInTransaction() && !isInBatch()) {
```
Comment (rajatbhatta) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java:1544]:
```suggestion
    if (isInternalMetadataQuery || (isAutocommit() && !isInTransaction() && !isInBatch())) {
```
Makes it clearer.

**Review by rajatbhatta — APPROVED** [review]
LGTM, apart from just one nit.

**Pull request merged**